README.md

```diff
@@ -143,177 +153,227 @@ func completionHandler(tokenResult: Result<Token, Error>) -> Void {
 
 ```swift
-extension ProductDetailViewController: PaymentCreatorControllerDelegate {
-  func paymentCreatorController(_ paymentCreatorController: PaymentCreatorController, didCreatePayment payment: Payment) {
-    dismissForm()
 
-    // Sends selected `Token` or `Source` to your server to create a charge, or a customer object.
+let allowedPaymentMethods = SourceType.availableByDefaultInThailand
+omiseSDK.presentChoosePaymentMethod(
+    from: self,
+    amount: paymentAmount,
+    currency: paymentCurrencyCode,
+    allowedPaymentMethods: allowedPaymentMethods,
+    forcePaymentMethods: true,
+    isCardPaymentAllowed: true, 
+    handleErrors: true,
+    delegate: self
+)
+
+```
+
+Then implement the delegate to receive the `Source` or `Token` object after the user has selected:
+
+```swift
+extension ProductDetailViewController: ChoosePaymentMethodDelegate {  
+  func choosePaymentMethodDidComplete(with source: Source) {
+    omiseSDK.dismiss {
+      print(source.id) // prints Source ID
+    }
+  }
+  
+  func choosePaymentMethodDidComplete(with token: Token) {
+    omiseSDK.dismiss {
+      print(token.id) // prints Token ID
+    }
   }
 
-  func paymentCreatorController(_ paymentCreatorController: PaymentCreatorController, didFailWithError error: Error) {
+  func choosePaymentMethodDidComplete(with error: Error) {
+    // Only called if we set `handleErrors = false`.
+
     let alertController = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
     let okAction = UIAlertAction(title: "OK", style: .cancel, handler: nil)
     alertController.addAction(okAction)
-    paymentCreatorController.present(alertController, animated: true, completion: nil)
 
-    // Only important if we set `handleErrors = false`.
-    // You can send errors to a logging service, or display them to the user here.
+    let vc = omiseSDK.presentedViewController ?? self
+    vc.present(alertController, animated: true, completion: nil)
   }
-
-  func paymentCreatorControllerDidCancel(_ paymentCreatorController: PaymentCreatorController) {
-    dismissForm()
+  
+  func choosePaymentMethodDidCancel() {
+      omiseSDK.dismiss()
   }
 }
 ```
 
 ### Authorizing payment
 
-Some payment methods require the customers to authorize the payment via an authorized URL. This includes [3-D Secure verification](https://docs.opn.ooo/fraud-protection#3-d-secure), [Internet Banking payment](https://docs.opn.ooo/internet-banking), [Alipay](https://docs.opn.ooo/alipay). The Opn Payments iOS SDK provides a built-in class to authorize payments.
+Some payment methods require customers to authorize the payment uaing an authorized URL. This includes [3-D Secure verification](https://docs.opn.ooo/fraud-protection#3-d-secure), [Internet Banking payment](https://docs.opn.ooo/internet-banking), and [Alipay](https://docs.opn.ooo/alipay). The Opn Payments iOS SDK provides a built-in class to authorize payments.
```

Comment:
uaing => using

Reply:
Good point. it's also called _authorize uri_ rather than authoriz**ed** UR**L**